tests/unit/test_agw_network_configurator.py

```diff
@@ -0,0 +1,1194 @@
+#!/usr/bin/env python3
+# Copyright 2022 Canonical Ltd.
+# See LICENSE file for licensing details.
+
+import unittest
+from unittest.mock import Mock, PropertyMock, call, mock_open, patch
+
+from magma_access_gateway_installer.agw_network_configurator import (
+    AGWInstallerNetworkConfigurator,
+)
+
+
+class TestAGWInstallerNetworkConfigurator(unittest.TestCase):
+    TEST_NETWORK_INTERFACES = ["eth0", "eth1"]
+
+    def test_given_agw_installer_network_configurator_when_configure_network_interfaces_is_called_and_no_configuration_has_been_done_yet_then_entire_network_configuration_flow_is_executed(  # noqa: E501
```

Comment:
We don't want to write tests that validate that private methods are actually called or not called. What we want to validate is that for example, the DNS was configured by checking that the file was written to `/etc/resolved.conf` or whatever's the directory. So the setup for most tests around this `configure_network_interfaces` method would be the same but what we validate would be different.

Reply:
Also, it's typically a design smell when there's tons of mocks for a single public method.